cmd/libsnap-confine-private/locking.c

```diff
@@ -104,11 +104,11 @@ static int sc_lock_generic(const char *scope, uid_t uid)
 		// The root user doesn't have a per-user mount namespace.
 		// Doing so would be confusing for services which use $SNAP_DATA
 		// as home, and not in $SNAP_USER_DATA.
-		sc_must_snprintf(lock_fname, sizeof lock_fname, "%s/%s.lock",
-				 sc_lock_dir, scope ? : "");
+		sc_must_snprintf(lock_fname, sizeof lock_fname, "%s.lock",
+				 scope ? : "");
 	} else {
-		sc_must_snprintf(lock_fname, sizeof lock_fname, "%s/%s.%d.lock",
-				 sc_lock_dir, scope ? : "", uid);
+		sc_must_snprintf(lock_fname, sizeof lock_fname, "%s.%d.lock",
+				 scope ? : "", uid);
```

Comment:
Maybe nice to have, you could update the debug line just below to make it:
```
debug("opening lock file: %s/%s", sc_lock_dir, lock_fname);
```
Not a blocker though.